mantidproject/mantid
# Fix "plot guess" leaving old guesses on plot

Fixed the *Plot guess* functionality in MuonAnalysis. This should work the same as it did with the old fitting UI: there should be only one guess curve, which auto-updates when the function or parameters are changed. "Remove guess" should do what it says and not plot another guess.

When multi fit mode (the new fitting UI) is on, need to remove any plotted guess before updating the function in the model, then plot it again if necessary. This is because the `PropertyHandler` that knows whether a guess is plotted is held by the `CompositeFunction` that is being replaced.

Unit tests added to cover new function.

**To test:**

- Open *Interfaces/Muon/Muon Analysis* and, on the Settings tab, tick "enable multiple fitting".
- Load some data on the Home tab, for example `MUSR00015190.nxs` (unit test data).
- Go to Data Analysis tab and set a fit function in the function browser at the top. Click *Display/Plot guess*.
- Edit one of the parameters of the function and verify the guess curve updates automatically.
- Click *Display/Remove guess* and verify the guess is removed.
- Plot the guess again, then run the fit and remove the guess. This should work OK and not plot a second guess when you try to remove it.

Turn off "enable multiple fitting", then close and reopen MuonAnalysis and follow the same steps. "Plot guess" should work here too (no change from before).

Fixes #18060.

[Release notes](https://github.com/mantidproject/mantid/blob/5fe0bc7b2b4c1d9287a537a18ab3303bab353bd6/docs/source/release/v3.9.0/muon.rst#id1) have been updated.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
